account_payment_order/models/account_payment_order.py

```diff
@@ -196,6 +196,26 @@ def check_date_scheduled(self):
                         )
                     )
 
+    @api.constrains("payment_line_ids")
+    def _check_payment_lines(self):
+        for order in self:
+            payment_lines = order.payment_line_ids
+            move_lines_ids = payment_lines.mapped("move_line_id").ids
+            existing_move_lines = []
+            for move_line_id in move_lines_ids:
+                existing_move_line = payment_lines.filtered(
+                    lambda l: l.move_line_id.id == move_line_id
+                )
+                if len(existing_move_line) > 1:
+                    existing_move_lines.append(existing_move_line[0].communication)
+            if existing_move_lines:
```

Comment:
Avoid `s` in the middle of the variable name, as they provoke often typos, and this is also very unperformant. You can avoid for this case to specify which is duplicated, as it's a very rough edge:

```suggestion
            move_lines = payment_lines.move_line_id
            if len(move_lines) != len(payment_lines):
```

Reply:
Thank you Pedro for your time.
I have changed `move_lines_ids` to `move_line_ids` as you suggested.
I consider necessary to specify which line is being duplicated, as some payment orders can have numerous lines, and it would be harder to find the duplicated lines for the user.

Reply:
But as said, this is anti-performant, with a quadratic complexity that cannot be accepted. If you don't find something with a linear complexity to get the same result, you should remove it.

Reply:
OK, it has been modified. Thank you @pedrobaeza.